### ISSUE TITLE
[REF-742] CTRL-C does not exit reflex in python 3.12

### ISSUE THREAD

**Issue by masenf** [reflex-dev/reflex#1916]
**To Reproduce**
```
python3.12 -m venv VENV
. ./VENV/bin/activate
pip install reflex
reflex init
reflex run
```

Then press CTRL-C...

**Expected behavior**
Expect that Reflex exits. Instead it just doesn't seem to respond at all.

**Specifics (please complete the following information):**
 - Python Version: 3.12.0
 - Reflex Version: 0.2.8
 - OS: macOS 13.5.2
 - Browser (Optional): N/A

<sub>[REF-742](https://linear.app/reflex-dev/issue/REF-742/ctrl-c-does-not-exit-reflex-in-python-312)</sub>

**Comment by masenf** [comment]
We seem to be getting stuck here: https://github.com/encode/uvicorn/blob/40b99b8436c0c261e3a85d10e291424072946292/uvicorn/server.py#L272

server._active_count is 2 after the sockets are closed, so i think this might be a uvicorn bug... if i move that `await server.wait_close()` line down below the `connection.shutdown()` block, then it reliably works

https://github.com/encode/uvicorn/blob/40b99b8436c0c261e3a85d10e291424072946292/uvicorn/server.py#L274-L277

**Comment by masenf** [comment]
Minimal repro

```python
import fastapi
import uvicorn

app = fastapi.FastAPI()


@app.get("/")
async def root():
    return {"message": "Hello World"}


uvicorn.run(
    app=app,
    host="127.0.0.1",
    port=8092,
)
```

```
python3.12 -m venv VENV-uvicorn312
./VENV-uvicorn312/bin/pip install uvicorn fastapi
./VENV-uvicorn312/bin/python run_app.py
```
access the app on http://localhost:8092, then press CTRL-C to exit the server.

**Comment by masenf** [comment]
Tracking potential upstream issue as https://github.com/encode/uvicorn/discussions/2122

**Comment by masenf** [comment]
Fixed via #2006
